packages/vizbuilder/src/helpers/sorting.js

```diff
@@ -0,0 +1,211 @@
+import sort from "fast-sort";
+import union from "lodash/union";
+import {unique} from "shorthash";
+
+import {STATES} from "./constants";
+import {sortSlice} from "./random";
+
+export function isTimeDimension(dimension) {
+  return dimension.dimensionType === 1 || dimension.name === "Date" || dimension.name === "Year";
+}
+
+export function injectCubeInfoOnMeasure(cubes) {
+  let nCbs = cubes.length;
+  while (nCbs--) {
+    const cube = cubes[nCbs];
+
+    if (cube.annotations.hide_in_ui) {
+      cubes.splice(nCbs, 1);
+      continue;
+    }
+
+    const cbName = cube.caption || cube.name;
+    const cbTopic = cube.annotations.topic;
+    const cbSubtopic = cube.annotations.subtopic;
+    const selectorKey = `${cbTopic}_${cbSubtopic}_`;
+    const sortKey = sortSlice(selectorKey);
+    const sourceName = cube.annotations.source_name;
+    // const sourceDesc = cube.annotations.source_description;
+    // const sourceLink = cube.annotations.source_link;
+    // const datasetName = cube.annotations.dataset_name;
+    // const datasetLink = cube.annotations.dataset_link;
+
+    cube.annotations._key = unique(cbName);
+
+    let nMsr = cube.measures.length;
+    while (nMsr--) {
+      const measure = cube.measures[nMsr];
+      const annotations = measure.annotations;
+
+      annotations._key = unique(`${cbName} ${measure.name}`);
+      annotations._cb_name = cbName;
+      annotations._cb_topic = cbTopic;
+      annotations._cb_subtopic = cbSubtopic;
+      annotations._cb_sourceName = sourceName;
+      annotations._selectorKey =
+        selectorKey + (measure.caption || measure.name);
+      annotations._sortKey = `${sortKey}${measure.caption ||
+        measure.name}`.toLowerCase();
+      // annotations._source_desc = sourceDesc;
+      // annotations._source_link = sourceLink;
+      // annotations._dataset_name = datasetName;
+      // annotations._dataset_link = datasetLink;
+    }
+
+    let nDim = cube.dimensions.length;
+    while (nDim--) {
+      const dimension = cube.dimensions[nDim];
+      const keyPrefix = `${cbName} ${dimension.name} `;
+
+      let nHie = dimension.hierarchies.length;
+      while (nHie--) {
+        const hierarchy = dimension.hierarchies[nHie];
+
+        let nLvl = hierarchy.levels.length;
+        while (nLvl--) {
+          const level = hierarchy.levels[nLvl];
+
+          level.annotations._key = unique(keyPrefix + level.name);
+        }
+      }
+    }
+  }
+}
+
+export function getValidMeasures(cubes) {
+  cubes = [].concat(cubes);
+  const measures = [];
+
+  let nCbs = cubes.length;
+  while (nCbs--) {
+    const cube = cubes[nCbs];
+    let nMsr = cube.measures.length;
+    while (nMsr--) {
+      const measure = cube.measures[nMsr];
+      // TODO: ensure this is checked server-side and remove
+      if (/\smoe$/i.test(measure.name)) {
+        continue;
+      }
+      const key = measure.annotations.error_for_measure;
```

Comment:
That's the reason for the `// TODO`
For test purposes, and since this annotation is not completely applied to the main cube server, this works as a safeguard.